### PR TITLE
Add support for --pathmap

### PR DESCRIPTION
**Pull request by saul**
# Superseded by #6609

Fixes https://github.com/Microsoft/visualfsharp/issues/5213, https://github.com/Microsoft/visualfsharp/issues/3812

### CODE REVIEW

File: src/utils/PathMap.fs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+/// Functions to map real paths to paths to be written to PDB/IL
+namespace Internal.Utilities
+
+open System
+open System.IO
+
+type PathMap = PathMap of Map<string, string>
+
+[<RequireQualifiedAccess>]
+module internal PathMap =
```
Comment (TD5) [inline src/utils/PathMap.fs]: Very clear model 👍

Comment (saul) [reply src/utils/PathMap.fs]: Thanks Tom - it's a shame that we don't have a better model for file systems. Then this map wouldn't be string to string - alas :)

**Comment by cartermp** [comment]
@brettfo @KevinRansom @dsyme can y'all take a look?

**Comment by saul** [comment]
Before merging, I’ll pull this up to master and add support for `<PathMap>` in project files, too.

**Comment by saul** [comment]
Superseded by #6609, closing this off.

**Pull request closed without merging**